packages/components/src/popover/index.tsx

```diff
@@ -416,10 +420,12 @@ const UnforwardedPopover = (
 	// document scrolls. Also update the frame offset if the view resizes.
 	useLayoutEffect( () => {
 		if (
-			// reference and floating are in the same document
+			// Reference and root documents are the same.
 			referenceOwnerDocument === document ||
-			// the reference's document has a view (i.e. window)
-			// and a frame element (ie. it's an iframe)
+			// Reference and floating are in the same document.
+			referenceOwnerDocument === floatingOwnerDocument ||
```

Comment:
Can't we just use the `refs.floating` from use `useFloating` hook?

Reply:
That one's a callback ref.

Reply:
The `floating` is the callback ref, I meant from `refs` [property](https://floating-ui.com/docs/react-dom#internal-refs). I quickly tried it and seems to work fine. Didn't test thoroughly though..

Reply:
Ah, who knew! (Nik did)

I'll update the PR.